### ISSUE TITLE
[Feature Request] Support for batch img2img?

### ISSUE THREAD

**Issue by Creative-Ataraxia** [Mikubill/sd-webui-controlnet#61]
how do we load the same image into controlnet that is loaded into img2img? So we dont' have to manually load it twice each time

**Comment by Mikubill** [comment]
Already supported by #36

**Comment by Creative-Ataraxia** [comment]
> Already supported by #36

thanks for the reply; What I meant was, not just the `img2img` tab, but the `batch` subtab of the `img2img` tab; when using the `batch` tab and the controlnet script, will raise error:

```\stable-diffusion-webui\modules\img2img.py", line 76, in process_batch
processed_image.save(os.path.join(output_dir, filename))
AttributeError: 'numpy.ndarray' object has no attribute 'save'```

sounds like the output data of controlnet is a `ndarray` and when processed as webui's code `processed_image`, by line 76, `processed_image.save(os.path.join(output_dir, filename))`, will raise an error

**Comment by ryanthegray74** [comment]
yes currently batch only gives you the very first image and does not work for the rest , this shouldnt be that hard to fix

**Comment by Creative-Ataraxia** [comment]
> yes currently batch only gives you the very first image and does not work for the rest , this shouldnt be that hard to fix

wonder if we can extract the for loop of the batch function and output the image into controlnet somehow

**Comment by Mikubill** [comment]
Added an option to skip saving controlnet detectmep. May help with this issue

**Comment by boredofnames** [comment]
Can confirm this is working now with "Do not append detectmap to output" enabled in settings. Thanks!

**Comment by toyxyz** [comment]
> Added an option to skip saving controlnet detectmep. May help with this issue

When I use "Do not append detectmap to output" Batch img2img works fine! In my personal opinion, this option would be more convenient to use if moved to the ControlNet panel of the img2img tab.

**Comment by Creative-Ataraxia** [comment]
> Added an option to skip saving controlnet detectmep. May help with this issue

great, thanks! just wondering, does this mean that the open pose are no longer guiding the generated images? I noticed that some some output images are not following the input image's pose anymore, with openpose selected

**Comment by ryanthegray74** [comment]
maybe adding more weight can fix that

**Comment by toyxyz** [comment]
Found something a little odd. BatchImg2Img runs normally without any problems, but compared to when ControlNet was run in Txt2img (same Seed, prompt), it seems that not only openpose is input, but the image itself is also input. 1: Input Image, 2: Txt2img Openpose, 3: BatchImg2Img Openpose. Is there a way to get the same result as Txt2Img in BatchImg2Img?
![image](https://user-images.githubusercontent.com/8006000/219093833-a85e499d-717c-4442-9e00-c0c47b835e58.png)

**Comment by ddPn08** [comment]
I created a temporary extension to fix this issue.
By using this extension, img2img processing will not be performed.
https://github.com/ddPn08/sd-webui-controlnet-batch-patch

Once #98 is merged it will no longer be needed.

**Comment by Creative-Ataraxia** [comment]
> I created a temporary extension to fix this issue. By using this extension, img2img processing will not be performed. https://github.com/ddPn08/sd-webui-controlnet-batch-patch
> 
> Once #98 is merged it will no longer be needed.

I double that current outputs are all really dark, which seems to be because it's using the detectmaps as the img2img sources, is this what your PR aims to fix?
